static/js/buddy_data.js

```diff
@@ -322,7 +369,37 @@ function get_user_id_list(user_filter_text) {
 
 exports.get_filtered_and_sorted_user_ids = function (user_filter_text) {
     let user_ids;
-    user_ids = get_user_id_list(user_filter_text);
+    let recipient_list = '';
+    let narrow_filter;
+
+    if (exports.should_show_only_recipients() && narrow_state.active()) {
+        narrow_filter = narrow_state.filter();
+        if (narrow_filter.has_operator('pm-with')) {
+            recipient_list = 'pm';
+        }
+        if (narrow_filter.has_operator('stream')) {
+            recipient_list = 'stream';
+        }
+    }
+
+    switch (recipient_list) {
+    case 'pm':
+        user_ids = get_pm_member_list(narrow_filter.operands('pm-with')[0].split(','));
+        if (user_filter_text) {
```

Comment:
We don't really need these `if (user_filter_text)` conditions--the function `filter_user_ids` does the right thing here.

What's our policy on not-recently-active users when we're in a narrow, @timabbott?